### ISSUE TITLE
Documentation functions work intermittently

### ISSUE THREAD

**Issue by mars0i** [liquidz/vim-iced#390]
I'm experiencing a strange error, and I can't report it properly because I can't localize the problem yet.    I can point to specific files that illustrate the problem in the repo I'm working on, but I don't know what is different about them.  I'm reporting the issue now because maybe you will have some suggestions about experiments to perform.    Thank you!

The problem seems to be this:

1. When I run `iced repl` with no arguments, and then connect from Vim, in *some* of my source files `IcedDocumentPopupOpen`, `IcedDocumentPopupShow`, and `IcedDocumentShow` *only* work on functions in clojure.core.  For other functions, including those defined in my source files and files in Clojure contrib libraries, I get the message "Not found."

2. But when I run `iced repl --instant` and connect, those vim-iced functions work on all function references in the files mentioned above.  However, now there are *other* source files in which `IcedDocumentPopupOpen`, `IcedDocumentPopupShow`, and `IcedDocumentShow` *only* work on functions in clojure.core.

Summary: No matter what, I get "Not found." for some function references in some of my source files--but not other source files--but whether I include `--instant` or not with `iced repl` changes which files have the problem.

(One theory I have is that there's some kind of race condition, and whether `--instant` is used speeds up or slows down processing of source files.)

**Comment by liquidz** [comment]
@mars0i Thanks for your reporting!
I'll have a look later.
If you have a minimal project to reproduce, please let me know.

**Comment by mars0i** [comment]
Thanks @liquidz .

Here is an example  project that is not very minimal, but it illustrate the problem--if the problem shows up on your system.  (It's so strange that I will not be surprised if you can't reproduce it easily.)

The attached zip file contains a tar file with a Leiningen repo.  You won't be able to run the source files under src/clj, but that's not necessary to see the problem.  (I can give you the full repo if you want.)

There are two source files, walks.clj and fileio.clj.  

When I go to the bottom of fileio.clj and type `K` or `<Leader>hs` on the name `spit-csv`, I get documentation and source, respectively.  This works on the other function names in this file.  

In walks.clj, if I go to the bottom and type `K` or `<Leader>hs` on `path-with-food` or `walk-stops` (both defined earlier in the file), I get the "Not found." error.  However, if I type `K` or `<Leader>hs` on `conj` or any other clojure.core function, they work.


[icedtest.zip](https://github.com/liquidz/vim-iced/files/8281345/icedtest.zip)

**Comment by mars0i** [comment]
I forgot to say that now the problem's occurring consistently in the same way (i.e. in walks.clj) whether I use `--instant` or not.  Except that in the original repo, if I start the repl with `IcedJackIn`, sometimes the problem occurs in fileio.clj.  I don't know.  It is weird.  

(I found two timeout variables in vim-iced.  At one point I tried increasing both of them to 10000 to see if that made a difference, but it didn't.)

**Comment by liquidz** [comment]
@mars0i Sorry for late reply.

In my environment, `[mason "20"]` in project.clj could not be found, and I failed to run `iced repl`
```
Could not find artifact mason:mason:jar:20 in central (https://repo1.maven.org/maven2/)
Could not find artifact mason:mason:jar:20 in clojars (https://repo.clojars.org/)
```

If successful, `walks.clj` requires `utils.math` and `utils.random` these are not contained to attached file, so I can't evaluate the file.

When I removed `[mason "20"]`, `utils.math` and `utils.random` to **force** to run `iced repl`, I can get documentation like follows.
<img width="763" alt="2022-03-19 6 46 26" src="https://user-images.githubusercontent.com/6941/159087615-f3d971cb-0029-4f6e-8244-05a053135c05.png">

### Environment

* vim 8.2.4286
* nvim v0.6.1
* vim-iced v3.8.2

### Steps to reproduce

* Run `iced repl`
* Run `vim src/clj/fileio.clj`
  * Execute `:IcedConnect`
  * Move the cursor to the bottom of buffer and type K on the name `spit-csv`
    * I got documentation.
  * Execute `:e src/clj/walks.clj`
    * Execute `IcedRequire` to evaluate current buffer.
    * Move the cursor to the bottom of buffer and type K on the name `walk-stops`
      * I got documentation.

**Comment by mars0i** [comment]
Sorry about the MASON dependency.  I should have removed that.  And I didn't realize that you had to be able to evaluate the file--I was trying not to burden you by sending a whole repo.   But now I understand:

OK!  So what I didn't realize is that each file has to be `require`d by `IcedRequire` before vim-iced will know about the definitions.  (hat is also why I thought I could give you a partial repo in which files were missing dependencies.  (But I should have realized, regardless, that you wouldn't be able to start a repl with that project file with the MASON reference.)

And now it works!  I'm happy.  This is great.  Thank you.

Part of what confused me is that I guess that the file that I'm in when I connect to the repl will automatically get `require`d, so its definitions will be available.  I was seeing that those definitions in the first file were available without me doing anything further, and the clojure.core definitions were available without doing anything further, but I didn't understand that I had to to run `IcedRequire` so that vim-iced could findi thfe other definitions automatically.  This also explains why sometimes I got definitions in one file and not the other, and sometimes it was the other way around: that depended on what file I was in when I connected to the repl.

I can see now that it makes sense that one has to tell the repl about each file using `IcedRequire` (or `<Leader>eb`): how else will it know about the definitions?  But there's a lot that vim-iced does that just seems like magic to me anyway, so I just figured it was magic. :-)

Again--thanks.  I'm very happy.
